packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -317,12 +316,30 @@ Iterable<Method> buildTags(
           toDartName(identifierBuilder.toString(), uppercaseFirstCharacter: true),
         );
 
-        code.writeln('''
-var _uri = Uri.parse(UriTemplate('${pathEntry.key}').expand(_pathParameters));
-if (_queryParameters.isNotEmpty) {
-  _uri = _uri.replace(queryParameters: _queryParameters);
-}
-''');
+        final queryParams = <String>[];
+        for (final parameter in parameters) {
+          if (parameter.$in != openapi.ParameterType.query) {
+            continue;
+          }
+
+          queryParams.add(parameter.uriTemplate(withPrefix: false));
+        }
+
+        final pathBuilder = StringBuffer()..write(pathEntry.key);
+
+        if (queryParams.isNotEmpty) {
+          pathBuilder
+            ..write('{?')
+            ..writeAll(queryParams, ',')
+            ..write('}');
+        }
+
+        final path = pathBuilder.toString();
+
+        // Sanity check the uri at build time.
+        UriTemplate(path);
```

Comment:
Exceptions should be caught and a better error message should be logged instead.

Reply:
I agree but how do we easily quit generation?
Just rethrow?

Reply:
Throw a new exception with the improved error message (and link to the rfc would be good as well).